src/roswire/ros2/node.py

```diff
@@ -5,20 +5,34 @@
 
 import attr
 
+from ..interface import Node
+
 if typing.TYPE_CHECKING:
     from ..app import AppInstance
 
 
 @attr.s(frozen=True, auto_attribs=True, slots=True)
-class ROS2Node:
+class ROS2Node(Node):
     """Provides an interface for interacting with a ROS2 node.
 
     Attributes
     ----------
     name: str
         The name of the node.
     """
-    app_instance: 'AppInstance'
+    app_instance: 'AppInstance' = attr.ib()
+    __name: str
```

Comment:
`__name` doesn't do anything here. You probably want to use `attr.ib()` to create a public `name` attribute. Notice that `frozen=True` appears in the `@attr.s` decorator for the class: That instructs attrs to create an immutable class. That is, users can read the `name` attribute but they can't write to it.